Source/Peripheral.swift

```diff
@@ -543,3 +547,25 @@ extension Peripheral: Equatable { }
 public func == (lhs: Peripheral, rhs: Peripheral) -> Bool {
     return lhs.peripheral == rhs.peripheral
 }
+
+fileprivate extension Peripheral {
+    fileprivate typealias `Self` = Peripheral
+
+    /**
+     Filters an item list based on the provided UUID list. Only items returned whose UUID matches an item in the provided UUID list.
+     Each UUID should have at least one item matching in the items list. Otherwise the result is nil.
+
+     - uuids: a UUID list or nil
+     - items: items to be filtered
+     - itemUUID: closure to extract the UUID from the items
+     - Returns: the filtered item list
+    */
+    static func filterCachedItems<T>(uuids: [CBUUID]?, items: [T], itemUUID: (T) -> CBUUID) -> [T]? {
```

Comment:
@Kacper20 Thank you for your feedback. I think your suggestion is a pretty good idea. I'm gonna implement the changes based on your comments and force push the updated PR.